### PR TITLE
Add post spawn hook

### PR DESCRIPTION
**Pull request by jamescurtin**
There is a `pre_spawn_hook` that is executed before the spawner starts; this PR implements a corresponding `post_spawn_hook` that is executed after the spawner stops.

### CODE REVIEW

File: jupyterhub/spawner.py [hunk 1]
```diff
@@ -515,6 +515,15 @@ def my_hook(spawner):
         """
     ).tag(config=True)
 
+    post_spawn_hook = Any(
```
Comment (minrk) [inline jupyterhub/spawner.py]: Can we call it `post_stop_hook`? I would expect `post_spawn` to be after start finishes (we probably should have called `pre_spawn` `pre_start`).

**Comment by minrk** [comment]
I made a comment about the name of the hook, but this makes sense.

**Comment by jamescurtin** [comment]
👍 I've updated the name of the hook. Thanks for the review!

**Comment by minrk** [comment]
Thanks!

**Pull request merged**